Add basic workflow for linux builds.

pmienk opened the pull request.

evoskuil reviewed (approved):
Looks awesome, the only comment I have is that in the interest of style consistency I would keep the generate4.xml <job> attributes on a single line.

pmienk commented:
> Looks awesome, the only comment I have is that in the interest of style consistency I would keep the generate4.xml attributes on a single line.

Corrected, though the lines rapidly surpass an 80 character limit.  Also fixed a mistaken optimization overlook (changed the xml, forgot to alter the logic).
